internal/backend/btree/node.go

```diff
@@ -0,0 +1,144 @@
+package btree
+
+import (
+	"encoding/binary"
+	"fmt"
+)
+
+/*
+	Node is the BTree logical implementation
+
+	order: btree order which is the same number of pointers
+	       in the node and the number of keys plus one
+	cells: key pointer pairs ordered by the key value
+*/
+type Node struct {
+	nodeType
```

Comment:
nodeType is a byte array right? should the type be dfined here?

Reply:
i can define the type - it's just a single byte which is like wrapped in a type called `nodeType`. This code is really confusing let me try to think of a way to make it clearer